backends/bmv2/common/action.cpp

```diff
@@ -47,6 +47,36 @@ void ActionConverter::convertActionBody(const IR::Vector<IR::StatOrDecl>* body,
         // TODO(jafingerhut) - add line/col at all individual cases below,
         // or perhaps it can be done as a common case above or below
         // for all of them?
+
+        IR::MethodCallExpression *mce2;
+        auto isR = false;
+        if (s->is<IR::AssignmentStatement>()) {
+            auto assign = s->to<IR::AssignmentStatement>();
+            const IR::Expression *l, *r;
+            l = assign->left;
+            r = assign->right;
+            auto mce = r->to<IR::MethodCallExpression>();
+            if (mce != nullptr) {
+                auto mi = P4::MethodInstance::resolve(mce, ctxt->refMap, ctxt->typeMap);
+                auto em = mi->to<P4::ExternMethod>();
+                if (em != nullptr) {
+                    if (em->originalExternType->name.name == "Register" ||
```

Comment:
In general we avoid using strings in the code, and have these strings be defined as constants in a model header file. I don't recall whether there's a psaModel.h file.

Reply:
The bmv2 backend is using `frontends/p4/fromv1.0/v1model.h`.  This header has definition for a register. 

There is no psaMode.h file in p4c just yet.  I have used `find ./ -name "*model*.h"` and `find ./ -name "*Model*.h"` to check. For now, we could add `#include "frontends/p4/fromv1.0/v1model.h"` to actions.cpp.

Reply:
Thanks! I found that there is a `psa.p4` file in p4c/p4include which is the model header file for psa. I will have the strings define in this file.

Reply:
Thanks! I will try to see whether there is a model header file for psa. If not, I can create one.

Reply:
Note p4c/p4include/psa.p4 is a PSA switch architecture file.  In this file, externs supported by PSA are defined.  The model file is a .h file, not a .p4 file.  As I have said above, the model file being used by simple_switch is v1model.h.  v1model.h is included by several .cpp programs in p4c/backends/bmv2.   See if existing v1model.h serves your code changes.  If not define a new one, say, psamodel.h.

Reply:
Creating psa.h from scratch may be a lot of work, this is not really necessary to merge this PR.